frontend/src/app/components/signup-page/signup-page.component.ts

```diff
@@ -1,25 +1,95 @@
-import { Component, OnInit } from '@angular/core';
-import { HttpResponse } from '@angular/common/http';
-import { AuthService } from 'src/app/services/auth.service';
+import { Component, OnDestroy, OnInit } from '@angular/core';
 import { Router } from '@angular/router';
+import { FormArray, FormBuilder, FormGroup, Validators } from '@angular/forms';
+import { AuthService } from 'src/app/services/auth.service';
+import { MyErrorStateMatcher } from 'src/app/helpers/default.error-matcher';
+import { COMMON, SECURITY_QUESTION_1, SECURITY_QUESTION_2, SECURITY_QUESTION_3 } from 'src/app/helpers/common.const';
+import { Subscription } from 'rxjs';
+import { MatSnackBar } from '@angular/material/snack-bar';
+import { SecurityQuestion, User } from 'src/app/models/user.model';
 
 @Component({
-  selector: 'app-signup-page',
-  templateUrl: './signup-page.component.html',
-  styleUrls: ['./signup-page.component.scss']
+    selector: 'app-signup-page',
+    templateUrl: './signup-page.component.html',
+    styleUrls: ['./signup-page.component.scss']
 })
-export class SignupPageComponent implements OnInit {
+export class SignupPageComponent implements OnInit, OnDestroy {
+    public registrationForm: FormGroup;
+    public matcher = new MyErrorStateMatcher();
```

Comment:
I would suggest to keep it general since the error state matcher can be used for other fields as well, not just password. For this PR in particular, it is only used for password but can be used for others in the future